sdk/storage/azure-storage-file-share/azure/storage/fileshare/_models.py

```diff
@@ -833,13 +1044,114 @@ def _from_generated(cls, generated):
         return copy
 
 
+class FileProperties(DictMixin):
+    """File's properties class."""
+
+    name: Optional[str] = None
```

Comment:
Same as directory, is this really Optional? I think it sort of is because it doesn't come from the service but if you look at `get_file_properties` function you will see we fille in the `name`.
https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/storage/azure-storage-file-share/azure/storage/fileshare/_file_client.py#L1094

You can also see on line 1129 below, we fille it in in `from_generated` (which is used during a list call). So while it is technically Optional, it will never be returned to the customer without a `name`. If I am not missing a case, I think the correct thing to is type this as non-optional so it's easier to use for customers.

Reply:
Agreed making `name` non-optional will be easier :)

Reply:
You covered all the grounds here too!